src/lib/x509/x509_ext.cpp

```diff
@@ -742,8 +744,14 @@ void CRL_Distribution_Points::decode_inner(const std::vector<uint8_t>& buf) {
    m_crl_distribution_urls.push_back(ss.str());
 }
 
-void CRL_Distribution_Points::Distribution_Point::encode_into(DER_Encoder& /*to*/) const {
-   throw Not_Implemented("CRL_Distribution_Points encoding");
+void CRL_Distribution_Points::Distribution_Point::encode_into(DER_Encoder& der) const {
+   der.start_sequence()
+      .start_cons(ASN1_Type(0), ASN1_Class::ContextSpecific)
+      .start_cons(ASN1_Type(0), ASN1_Class::ContextSpecific)
+      .add_object(ASN1_Type(6), ASN1_Class::ContextSpecific, m_point.get_first_attribute("URI"))
+      .end_cons()
+      .end_cons()
+      .end_cons();
```

Comment:
This doesn't behave correctly if the distribution point doesn't contain a URL. (We'd encode an empty URL)

I'm not sure what the best fix is. The simplest thing to do would be to check that the distribution point alt name does contain a URL; if it does, do what you've done here. Otherwise, throw `Not_Implemented`. That would be fine with me and would at least make this more functional than the current state.

Also I suppose this doesn't behave correctly if the alternative name includes multiple URLs; we'd just encode the first. That would be actively bad since someone would think their distribution points were encoded but they were not. This could be fixed by either checking that there is *exactly* 1 URL (and throwing `Not_Implemented` if otherwise) or by iterating and encoding all of the URLs.

Reply:
Fully agree, I've reworked the method addressing your concerns. It should now accommodate these cases.